tests/functional/cli/admin/test_item_check.py

```diff
@@ -0,0 +1,1357 @@
+        self.container = "item_check_container" + random_str(4)
+        self.obj_name = "item_check_obj_" + random_str(4)
+        self.check_opts = self.get_opts(['Type', 'Item', 'Status'])
+
+        self.beanstalkd0.drain_tube('oio-preserved')
+
+        self.account_client = AccountClient(self.conf)
+        self.blob_client = BlobClient(self.conf)
+        self.autocontainer = HashedContainerBuilder(bits=self.FLAT_BITS)
+
+    def assertCheckOutput(self, expected_items, actual_output):
+        super(ItemCheckTest, self).assertListEqual(
+            sorted(expected_items),
+            sorted(actual_output.rstrip('\n').split('\n')))
+
+    def _wait_events(self, account, container, obj_name):
+        self.wait_for_event(
+            'oio-preserved',
+            fields={'account': account, 'user': container, 'path': obj_name},
+            type_=EventTypes.CONTENT_NEW)
+        self.wait_for_event(
+            'oio-preserved',
+            fields={'account': account, 'user': container},
+            type_=EventTypes.CONTAINER_STATE)
+
+    def create_object(self, account, container, obj_name):
+        opts = self.get_opts(['Name'])
+        output = self.openio(
+            '--oio-account %s object create %s /etc/passwd --name %s %s'
+            % (account, container, obj_name, opts))
+        self.assertOutput('%s\n' % obj_name, output)
+        opts = self.get_opts([], format='json')
+        output = self.openio(
+            '--oio-account %s object show %s %s %s'
+            % (account, container, obj_name, opts))
+        obj_meta = json.loads(output)
+        opts = self.get_opts(['Id'])
+        output = self.openio(
+            '--oio-account %s object locate %s %s %s'
+            % (account, container, obj_name, opts))
+        obj_chunks = output.rstrip('\n').split('\n')
+        self._wait_events(account, container, obj_name)
+        return obj_meta, obj_chunks
+
+    def create_object_auto(self, account, obj_name):
+        opts = self.get_opts(['Name'])
+        output = self.openio(
+            '--oio-account %s object create /etc/passwd --name %s '
+            '--auto --flat-bits %d %s'
+            % (account, obj_name, self.FLAT_BITS, opts))
+        self.assertOutput('%s\n' % obj_name, output)
+        opts = self.get_opts([], format='json')
+        output = self.openio(
+            '--oio-account %s object show %s --auto --flat-bits %d %s'
+            % (account, obj_name, self.FLAT_BITS, opts))
+        obj_meta = json.loads(output)
+        opts = self.get_opts(['Id'])
+        output = self.openio(
+            '--oio-account %s object locate %s --auto --flat-bits %d %s'
+            % (account, obj_name, self.FLAT_BITS, opts))
+        obj_chunks = output.rstrip('\n').split('\n')
+        container = self.autocontainer(obj_name)
+        self._wait_events(account, container, obj_name)
+        return container, obj_meta, obj_chunks
+
+    def corrupt_chunk(self, chunk):
```

Comment:
Should we add also `corrupt_chunk_xattr` ?
(Maybe it is done in another PR ?)

Reply:
I agree we are doing this test in another PR.